tle/cogs/acd_ai.py

```diff
@@ -0,0 +1,185 @@
+from discord.ext import commands
+from collections.abc import AsyncIterable
+import discord
+import google.generativeai as genai
+from tle.util import settings
+import requests
+from PIL import Image
+from tle import constants
+from ratelimit import limits, RateLimitException, sleep_and_retry
+
+
+
+ONE_MINUTE = 60
+MAX_CALLS_PER_MINUTE = 60
+API_KEY = constants.API_KEY
+
+image_types = ["jpeg", "png", "webp", "heic", "heif"]
+no_response_message = "Sorry, can't answer that. possible reasons might be no response, recitation, safety issue or blocked content."
+help_text = "```\nUsage:\n$chat : to create a public thread to interact with the bot.\n$private : to create a private thread to interact with the bot.\n\nACD AI answers to all text, image and attachment.\nHave fun!\n\nfor Admin/Mods:\n$set : to set a channel.\n$unset : to unset a channel.\n$show : to show current channels.\n$clear : to delete all the threads in set channels.```"
+allowed_guilds = [1063393625049923687, 501032525284769842]
+dev_id = 501026469569363988
```

Comment:
are these parameters supposed to change? if not why are we not declaring them as constants? (capitalise these).

Reply:
Will fix.